File: bindings/nodejs/lib/types/block/payload/payload.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

/**
 * All of the block payload types.
 */
enum PayloadType {
    /** A tagged data payload. */
    TaggedData = 0,
    /** A signed transaction payload. */
    SignedTransaction = 1,
    /** A candidacy announcement payload. */
    CandidacyAnnouncement = 2,
}

/**
 * The base class for block payloads.
 */
abstract class Payload {
    /**
     * The type of payload.
     */
    readonly type: PayloadType;

    /**
     * @param type The type of payload.
     */
    constructor(type: PayloadType) {
        this.type = type;
    }
}

export { PayloadType, Payload };

File: sdk/src/types/block/rand/payload.rs
// Copyright 2020-2021 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use crate::types::block::{
    payload::{tagged_data::TaggedDataPayload, Payload},
    rand::{bytes::rand_bytes, number::rand_number_range},
};

/// Generates a random tagged data payload.
pub fn rand_tagged_data_payload() -> TaggedDataPayload {
    TaggedDataPayload::new(
        rand_bytes(rand_number_range(TaggedDataPayload::TAG_LENGTH_RANGE).into()),
        rand_bytes(rand_number_range(0..10000)),
    )
    .unwrap()
}

/// Generates a random payload for a block.
pub fn rand_payload_for_block() -> Payload {
    // TODO complete
    rand_tagged_data_payload().into()
}
